src/Darryldecode/Cart/Cart.php

```diff
@@ -641,11 +640,15 @@ protected function validate($item)
      */
     protected function addRow($id, $item)
     {
+        $this->events->fire($this->getInstanceName().'.adding', array($item, $this));
+
         $cart = $this->getContent();
 
         $cart->put($id, new ItemCollection($item));
 
         $this->save($cart);
+
+        $this->events->fire($this->getInstanceName().'.added', array($item, $this));
     }
 
     /**
```

Comment:
Yes it is. The only thing that has changed is the first parameter of the event, which was first only the item object but can now be also an array with changeable data.